### PR TITLE
Quickstart tutorial (soft launch)

### PR DESCRIPTION
**Pull request by DavidBoike**
Connects to https://github.com/Particular/DeveloperEducation/issues/910

### CODE REVIEW

File: tutorials/quickstart/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+As shown in this diagram, the **ClientUI** endpoint, which serves as a stand-in for a web application in a more realistic example, will send a **PlaceOrder** command to the **Sales** endpoint. As a result, the **Sales** endpoint will publish an **OrderPlaced** event using the publish/subscribe pattern, which will be received by the **Billing** endpoint.
+
+![Initial Solution](before.svg)
+
+The solution mimics a real-life retail system, where the command to place an order occurs as a result of a customer interaction, and the actual processing occurs in the background. By publishing an event, the code to bill the credit card is isolated from the code to place the order, reducing coupling and making the system easier to maintain over the long term. Later in this tutorial, we'll see how to add a second subscriber in a **Shipping** endpoint which would begin the process of shipping the order.
+
```
Comment (weralabaj) [inline tutorials/quickstart/tutorial.md]: may be worth linking here to some "command vs event" article

File: tutorials/quickstart/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+In the **ClientUI** application, press `P` to place an order, and watch how the other windows react. It may happen too quickly for you to see, but **PlaceOrder** will be sent to **Sales** resulting in:
+
+    INFO  Sales.PlaceOrderHandler Received PlaceOrder, OrderId = 9b16a5ce-e6ae-4447-a911-b7d6e265a1f0
+
+As a result, the **OrderPlaced** event will be published, and the **Billing** window you will see:
+
+    INFO  Billing.OrderPlacedHandler Billing has received OrderPlaced, OrderId = 9b16a5ce-e6ae-4447-a911-b7d6e265a1f0
+
+Try pressing `P` repeatedly in the **ClientUI** window and watch the messages flow between endpoints.
+
```
Comment (weralabaj) [inline tutorials/quickstart/tutorial.md]: good place to introduce/mention SI visualization?

File: tutorials/quickstart/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+By using automatic retries, you can avoid losing data or having your system left in an inconsistent state on account of a stray database deadlock. No more spelunking through the database trying to fix business processes gone wrong!
+
+
+## Easy to extend
+
+Stub: Instructions for adding Shipping service with subscriber for OrderPlaced
+
+## Summary
+
+Summarize, link to CTA
```
Comment (weralabaj) [inline tutorials/quickstart/tutorial.md]: not sure if it's a good idea to link to blog from here, but "but all my errors are severe" would be excellent here

File: tutorials/quickstart/tutorial.md (hunk 1 above)
Comment (DavidBoike) [inline tutorials/quickstart/tutorial.md]: :+1: Added

File: tutorials/quickstart/tutorial.md (hunk 2 above)
Comment (DavidBoike) [inline tutorials/quickstart/tutorial.md]: Don't want to derail with explanation of how to install it.

Comment (weralabaj) [inline tutorials/quickstart/tutorial.md]: I understand, I was thinking more about "you can also visualize messages flow using SI(link)", nothing more.

**Comment by weralabaj** [comment]
@DavidBoike given that there'll be a lot of content to review, could we do that on a lesson by lesson basis? then we can merge a few prs one after another.

**Comment by DavidBoike** [comment]
@weralabaj My intent is for this tutorial to only have one part. No 5 lessons thing here. So the first draft of _all_ the text is now done.

**Comment by weralabaj** [comment]
@DavidBoike I've reviewed. One outstanding issue IMO is the code, i.e. the projects at the moment use core 6.3.0, but if you simply execute the command to add NServiceBus nuget it'll automatically use the latest (i.e. 6.3.2). Also should we explicitly mention that the project should target .Net framework 4.6.1 or higher?
I think by default it'll be 4.5.2 (at least it was for me).

File: tutorials/quickstart/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+Let's consider more carefully what happened. First, we had two processes communicating with each other with very little ceremony. The communication didn't break down even when the **Billing** service was unavailable. If we had implemented **Billing** as a REST endpoint, the **Sales** service would have thrown an HTTP exception when it was unable to communicate with it and *that request would have been lost*. By using NServiceBus we get a guarantee that even if message processing endpoints are temporarily unavailable, every message will eventually get delivered and processed.
+
+
+## Transient failures
+
+Have you ever business processes getting interrupted by transient errors like database deadlocks? Transient errors often leave a system in an inconsistent state, e.g. the order could be persisted in the database but not yet submitted to the payment processor. In such situation you might have to investigate the database like a forensic analyst, trying to figure out where the process went wrong, and how to manually jump-start it so that the process can complete.
```
Comment (kbaley) [inline tutorials/quickstart/tutorial.md]: Is _transient errors_ a term we can use without explanation or linking?

Comment (kbaley) [inline tutorials/quickstart/tutorial.md]: > In such situation

Should be:

> In such a situation

**Comment by DavidBoike** [comment]
@kbaley On "transient failure" ... I *think* it should be OK? I would hope that the words standing on their own (adverb + very well understood noun) should make it pretty clear especially in context.

**Comment by DavidBoike** [comment]
Updated to latest NSB, edited for comments from @kbaley and some things I noticed.

File: tutorials/quickstart/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+reviewed: 2017-06-05
+summary: In this tutorial, we'll see why software systems built on asynchronous messaging using NServiceBus are superior to traditional synchronous HTTP-based web services.
+extensions:
+- !!tutorial
+  downloadAtTop: true
+---
+
+In this tutorial, we'll see why software systems built on asynchronous messaging using NServiceBus are superior to traditional synchronous HTTP-based web services. We'll also show how NServiceBus guarantees reliability and extensibility that can't be achieved with REST.
+
+This tutorial skips over some concepts and implementation details in order to get up and running quickly. If you'd prefer a more in-depth tutorial, check out [Introduction to NServiceBus](/tutorials/intro-to-nservicebus/). It will teach you the NServiceBus API and important concepts you need to learn to build successful message-based software systems.
```
Comment (udidahan) [inline tutorials/quickstart/tutorial.md]: > check out Introduction to NServiceBus]

better with an "our" or "the" before "Introduction":

> check out our Introduction to NServiceBus

Did you also want to include "tutorial" at the end of that like you did at the end of this post?

File: tutorials/quickstart/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+
+
+## Transient failures
+
+Have you ever had business processes get interrupted by transient errors like database deadlocks? Transient errors often leave a system in an inconsistent state. For example, the order could be persisted in the database but not yet submitted to the payment processor. In such a situation you might have to investigate the database like a forensic analyst, trying to figure out where the process went wrong, and how to manually jump-start it so that the process can complete.
+
+With NServiceBus we don't need manual interventions. If an exception is thrown, then the message handler will automatically retry processing it. That addresses transient failures like database deadlocks, connection issues across machines, conflicts when accessing file to write, etc.
```
Comment (udidahan) [inline tutorials/quickstart/tutorial.md]: > With NServiceBus we don't need manual interventions

In the previous paragraph you used "you" language, but here you've switched to "we" language. Better to be consistent when you're so close to the same content.

File: tutorials/quickstart/tutorial.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+
+We saw how automatic retries provide protection from transient failures like database deadlocks. If we implement a multi-step process as a series of message handlers, then each step will be executed independently and can be automatically retried in case of failures. This means that a stray exception won't abort an entire process, leaving the system in an inconsistent state.
+
+We also implemented an additional event subscriber, showing how to decouple independent bits of business logic from each other. The ability to publish one event and then implement resulting steps in separate message handlers makes the system much easier to maintain and evolve.
+
+If you'd like to learn more about NServiceBus, check out the [Introduction to NServiceBus](/tutorials/intro-to-nservicebus/) tutorial. In it, you'll learn how to build the same `RetailDemo` solution from scratch, while learning the messaging concepts you'll need to know to build complex software systems with NServiceBus.
```
Comment (udidahan) [inline tutorials/quickstart/tutorial.md]: > you'll need to know to build complex software systems with NServiceBus.

Slightly better as:

> you'll need to know to build even more complex software systems with NServiceBus.

**Pull request merged**